### DISCUSSION TITLE
keyboard as device in yarp

### DISCUSSION THREAD

**Discussion by mhubii** [robotology/community#289] [Q&A] [answered]
Hi and thanks for taking the time,

is there a way to use a keyboard as a device in yarp, such that the robot can for example move by pressing w, a, s, and d? I would need some kind of interface that reads the keyboard input to a port. Is there a way to go?

Martin

**Comment by mhubii** [comment]
Okay sry I am pretty new to yarp. To solve this problem, one can implement a usual keyboard input via ncurses on linux and stream the commands using ports. If someone else is interested in how to implement this, check out my heicub_walking repository, it will be uploaded there soon.

Martin

**Comment by randaz81** [comment, marked as answer]
wait, you don't need to develop anything new! You can use the joystick driver already provided in YARP via SDL. The name of the device is SDLJoypad. 

Before starting it, just launch your preferred keyboard to joypad emulator.

**Comment by drdanz** [comment]
@randaz81 this is quite interesting and useful, is it documented somewhere?
